lib/wallet/http.js

```diff
@@ -837,6 +784,501 @@ class HTTP extends Server {
       await req.wallet.resend();
       res.json(200, { success: true });
     });
+
+    // Wallet Name States
+    this.get('/wallet/:id/name', async (req, res) => {
+      const height = this.wdb.height;
+      const network = this.network;
+
+      const names = await req.wallet.getNames();
+      const items = [];
+
+      for (const ns of names)
+        items.push(ns.getJSON(height, network));
+
+      res.json(200, items);
+    });
+
+    // Wallet Name State
+    this.get('/wallet/:id/name/:name', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+
+      assert(name, 'Must pass name.');
+      assert(rules.verifyName(name), 'Must pass valid name.');
+
+      const height = this.wdb.height;
+      const network = this.network;
+      const ns = await req.wallet.getNameStateByName(name);
+
+      res.json(200, ns.getJSON(height, network));
+    });
+
+    // Wallet Auctions
+    this.get('/wallet/:id/auction', async (req, res) => {
+      const height = this.wdb.height;
+      const network = this.network;
+
+      const names = await req.wallet.getNames();
+      const items = [];
+
+      for (const ns of names) {
+        const bids = await req.wallet.getBidsByName(ns.name);
+        const reveals = await req.wallet.getRevealsByName(ns.name);
+
```

Comment:
nit: how do you feel about removing this new line and adding one after creating `info`.

Reply:
I agree that looks better, pushed up a fix